lib/iris/tests/unit/coords/test_AuxCoord.py

```diff
@@ -0,0 +1,764 @@
+
+    def test_lazy_bounds_realise(self):
+        coord = AuxCoord(self.pts_real, bounds=self.bds_lazy)
+        real_bounds = coord.bounds
+        result = coord.core_bounds()
+        self.assertIs(result, real_bounds)
+
+
+@skip_lazy_auxcoords
+class Test_lazy_points(tests.IrisTest):
+    def setUp(self):
+        setup_test_arrays(self)
+
+    def test_real_core(self):
+        coord = AuxCoord(self.pts_real)
+        result = coord.lazy_points()
+        self.assertIsInstance(result, da.Array)
+        self.assertArrayEqual(result.dtype, self.pts_real.dtype)
+        self.assertArrayEqual(result.compute(), self.pts_real)
+
+    def test_lazy_core(self):
+        coord = AuxCoord(self.pts_lazy)
+        result = coord.lazy_points()
+        self.assertIs(result, self.pts_lazy)
+
+
+@skip_lazy_auxcoords
+class Test_lazy_bounds(tests.IrisTest):
+    def setUp(self):
+        setup_test_arrays(self)
+
+    def test_real_core(self):
+        coord = AuxCoord(self.pts_real, bounds=self.bds_real)
+        result = coord.lazy_bounds()
+        self.assertIsInstance(result, da.Array)
+        self.assertArrayEqual(result.dtype, self.bds_real.dtype)
+        self.assertArrayEqual(result.compute(), self.bds_real)
+
+    def test_lazy_core(self):
+        coord = AuxCoord(self.pts_real, bounds=self.bds_lazy)
+        result = coord.lazy_bounds()
+        self.assertIs(result, self.bds_lazy)
+
+
+@skip_lazy_auxcoords
+class Test_has_lazy_points(tests.IrisTest):
+    def setUp(self):
+        setup_test_arrays(self)
+
+    def test_real_core(self):
+        coord = AuxCoord(self.pts_real)
+        result = coord.has_lazy_points()
+        self.assertFalse(result)
+
+    def test_lazy_core(self):
+        coord = AuxCoord(self.pts_lazy)
+        result = coord.has_lazy_points()
+        self.assertTrue(result)
+
+    def test_lazy_core_realise(self):
+        coord = AuxCoord(self.pts_lazy)
+        coord.points
+        result = coord.has_lazy_points()
+        self.assertFalse(result)
+
+
+@skip_lazy_auxcoords
+class Test_has_lazy_bounds(tests.IrisTest):
+    def setUp(self):
+        setup_test_arrays(self)
+
+    def test_real_core(self):
+        coord = AuxCoord(self.pts_real, bounds=self.bds_real)
+        result = coord.has_lazy_bounds()
+        self.assertFalse(result)
+
+    def test_lazy_core(self):
+        coord = AuxCoord(self.pts_real, bounds=self.bds_lazy)
+        result = coord.has_lazy_bounds()
+        self.assertTrue(result)
+
+    def test_lazy_core_realise(self):
+        coord = AuxCoord(self.pts_real, bounds=self.bds_lazy)
+        coord.bounds
+        result = coord.has_lazy_bounds()
+        self.assertFalse(result)
+
+
+@skip_lazy_auxcoords
+class Test_bounds_dtype(tests.IrisTest):
+    def test_i16(self):
+        test_dtype = np.int16
+        coord = AuxCoord([1], bounds=np.array([[0, 4]], dtype=test_dtype))
+        result = coord.bounds_dtype
+        self.assertTrue(result == test_dtype)
```

Comment:
This looks ripe for being re-cast to `self.assertEqual(result, test_dtype)`... is there a reason you chose not to?